cdap-ui/app/cdap/components/DataPrepHome/index.js

```diff
@@ -150,13 +151,27 @@ export default class DataPrepHome extends Component {
             DataPrepStore.dispatch({
               type: DataPrepActions.disableLoading
             });
+            DataPrepStore.dispatch({
+              type: DataPrepActions.setWorkspaceList,
+              payload: {
+                list: []
+              }
+            });
             return;
           }
           let sortedWorkspace = orderBy(res.values, [(workspace) => workspace.name.toLowerCase()], ['asc']);
+          DataPrepStore.dispatch({
+            type: DataPrepActions.setWorkspaceList,
+            payload: {
+              list: sortedWorkspace
+            }
+          });
           let isCurrentWorkspaceIdValid = sortedWorkspace.find(ws => ws.id === this.props.match.params.workspaceId);
           if (this.props.match.params.workspaceId && !isCurrentWorkspaceIdValid) {
             let url = this.props.match.url.slice(0, this.props.match.url.indexOf(this.props.match.params.workspaceId));
             this.props.history.replace(url);
+          } else {
+            setWorkspace(sortedWorkspace[0].id).subscribe();
```

Comment:
So with this change, when we're on another tab and we click on 'Preparation', UI will show the first tab right? Should we still stay in the same tab instead?

Reply:
No we can't. The current url will have the workspace id but the link in the navbar will not. If the user clicks on navbar we should just go to the first workspace we get from backend in that order. If we want to retain the last accessed workspace it might be out of scope of this PR.